src/coreclr/src/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -735,6 +735,24 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                 GetEmitter()->emitIns_R_R_R(ins, emitSize, targetReg, op1Reg, op2Reg, opt);
                 break;
 
+            case NI_AdvSimd_Arm64_AddSaturateScalar:
+                if (varTypeIsUnsigned(node->GetAuxiliaryType()) != varTypeIsUnsigned(intrin.baseType))
```

Comment:
It might be nice to have a getter for `baseType` as well (though perhaps that also needs a better name).
Nothing that needs addressing now, though.